src/components/tooltip/Tooltip.tsx

```diff
@@ -138,16 +142,17 @@ class Tooltip extends React.Component<TooltipProps, State> {
         if (isControlled) {
             if (!prevProps.isShown && this.props.isShown) {
                 this.setState({ wasClosedByUser: false });
-            }
-        } else {
-            if (!prevState.isShown && this.state.isShown) {
-                // capture event so that tooltip closes before any other floating components that can be closed by
-                // "Escape" key(e.g. Modal, Menu, etc.)
                 document.addEventListener('keydown', this.handleKeyDown, true);
             }
-            if (prevState.isShown && !this.state.isShown) {
-                document.removeEventListener('keydown', this.handleKeyDown, true);
-            }
+        }
+
+        if (!prevState.isShown && this.state.isShown) {
+            // capture event so that tooltip closes before any other floating components that can be closed by
+            // "Escape" key(e.g. Modal, Menu, etc.)
+            document.addEventListener('keydown', this.handleKeyDown, true);
+        }
```

Comment:
moving this outside of the `else` would add a duplicate event listener if it also enters the above `if` block